mls-rs/examples/api_1x.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// Copyright by contributors to this project.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+use std::convert::Infallible;
+
+use mls_rs::{
+    client_builder::MlsConfig,
+    error::MlsError,
+    identity::{
+        basic::{BasicCredential, BasicIdentityProvider},
+        SigningIdentity,
+    },
+    CipherSuite, CipherSuiteProvider, Client, CryptoProvider, ExtensionList, KeyPackageStorage,
+};
+use mls_rs_core::key_package::KeyPackageData;
+
+const CIPHERSUITE: CipherSuite = CipherSuite::CURVE25519_AES128;
+
+fn main() -> Result<(), MlsError> {
+    let crypto_provider = mls_rs_crypto_openssl::OpensslCryptoProvider::default();
+
+    // Create clients for Alice and Bob
+    let alice = make_client(crypto_provider.clone(), "alice")?;
+    let bob = make_client(crypto_provider.clone(), "bob")?;
+
+    // Bob generates key package. We store secrets in memory, no need for any storage.
+    let key_package_generation = bob
+        .key_package_builder(CIPHERSUITE)?
+        .valid_for_sec(123)
+        .build()?;
+
+    let stored_secrets = key_package_generation.key_package_data;
+
+    // Alice creates a group with Bob.
+    let mut alice_group = alice.create_group(ExtensionList::default(), Default::default())?;
+
+    let welcomes = alice_group
+        .commit_builder()
+        .add_member(key_package_generation.key_package_message)?
+        .build()?
+        .welcome_messages;
+
+    alice_group.apply_pending_commit()?;
+
+    // Bob joins
+    let mut bob_group = bob.group_joiner(&welcomes[0], stored_secrets)?.join()?.0;
```

Comment:
[nit] Idk why group_joiner sounds weird, maybe we should think that through again